plugins/modules/dns_auth_zone.py

```diff
@@ -0,0 +1,2316 @@
+                        - "Defaults to empty."
+                    type: str
+                    returned: Always
+                use_default_mname:
+                    description:
+                        - "Optional. Use default value for master name server."
+                        - "Defaults to true."
+                    type: bool
+                    returned: Always
+"""  # noqa: E501
+
+from ansible_collections.infoblox.bloxone.plugins.module_utils.modules import BloxoneAnsibleModule
+
+try:
+    from bloxone_client import ApiException, NotFoundException
+    from dns_config import AuthZone, AuthZoneApi
+except ImportError:
+    pass  # Handled by BloxoneAnsibleModule
+
+
+class AuthZoneModule(BloxoneAnsibleModule):
+    def __init__(self, *args, **kwargs):
+        super(AuthZoneModule, self).__init__(*args, **kwargs)
+
+        exclude = ["state", "csp_url", "api_key", "id"]
+        self._payload_params = {k: v for k, v in self.params.items() if v is not None and k not in exclude}
+        self._payload = AuthZone.from_dict(self._payload_params)
+        self._existing = None
+
+    @property
+    def existing(self):
+        return self._existing
+
+    @existing.setter
+    def existing(self, value):
+        self._existing = value
+
+    @property
+    def payload_params(self):
+        return self._payload_params
+
+    @property
+    def payload(self):
+        return self._payload
+
+    def payload_changed(self):
+        if self.existing is None:
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = AuthZoneApi(self.client).read(self.params["id"], inherit="full")
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "absent":
+                    return None
+                raise e
+        else:
+            filter = f"fqdn=='{self.params['fqdn']}' and primary_type=='{self.params['primary_type']}'"
+            resp = AuthZoneApi(self.client).list(filter=filter, inherit="full")
+            if len(resp.results) == 1:
+                return resp.results[0]
+            if len(resp.results) > 1:
+                self.fail_json(msg=f"Found multiple AuthZone: {resp.results}")
+            if len(resp.results) == 0:
+                return None
+
+    def create(self):
+        if self.check_mode:
+            return None
+
+        resp = AuthZoneApi(self.client).create(body=self.payload, inherit="full")
+        return resp.result.model_dump(by_alias=True, exclude_none=True)
+
+    def update(self):
+        if self.check_mode:
+            return None
+
+        update_body = self.payload
+        update_body.fqdn = None
+        update_body.primary_type = None
```

Comment:
is this because you cannot update the fqdn and primary_type in a PATCH call? 

did you test what happens when a user tries to update the primary_type, does the user get an error?

for eg. the following scenario
- domain.com, cloud
- domain.com, external

I believe because of your filter condition, it will try to fetch (domain.com, external), but it will return nothing so will be a create call instead of update. The create call will however fail because you can't have two names with the same fqdn.